pyxrf/xanes_maps/xanes_maps_api.py

```diff
@@ -2934,6 +3035,60 @@ def _save_xanes_maps_to_tiff(*, wd, eline_data_aligned, eline_selected,
             print(f"\nR-factors for XANES maps are saved to file '{fln_xanes_rfactor}'.", file=f_log)
 
 
+def _save_spectrum_as_csv(*, fln, wd=None, msg_info=None, energy=None, spectrum=None):
+    """
+    Saves spectrum as CSV file.
+
+    Parameters
+    ----------
+
+    fln: str
+        name of the file to save spectrum
+
+    wd: str
+        working directory where the file ``file`` is to be placed
+
+    msg_info: str
+        A string that will be placed at the beginning of the CSV file before the file header.
+        Each line in the string must start with # (to separate the comment).
+
+    energy: ndarray(float)
+        The array of energy values in keV
+
+    spectrum: ndarray(float)
+        The array of spectrum values. Must have the same size as 'energy'.
+    """
+
+    # Full file path
+    file_path = os.path.join(wd, fln) if wd else fln
+
+    try:
+        # Verify that the arrays are valid and provide user-friendly message
+        if energy is None:
+            raise ValueError("The array 'energy' is None")
+        if spectrum is None:
+            raise ValueError("The array 'spectrum' is None")
+        if len(energy) != len(spectrum):
+            raise ValueError(f"Arrays 'energy' and 'spectrum' have different size: "
+                             f"len(energy)={len(energy)} len(spectrum)={len(spectrum)}")
+        # Create 2D array with 'energy' and 'spectrum' as columns
+        energy = np.asarray(energy)
+        spectrum = np.asarray(spectrum)
+        data = np.transpose(np.vstack((energy, spectrum)))
+        data = pd.DataFrame(data, columns=["Incident Energy, keV", "XANES spectrum"])
+        # Format as 'CSV'
+        str = data.to_csv(index=False)
+        # Attach the comment to the beginning of the formatted string
+        str = f"{msg_info}\n{str}"
```

Comment:
Can we add some kind of validation that the `msg_info` string has the `#` symbols for every line, and add them if users forget to add them? This would help to make sure that invalid files are not produced. Also, having a test for readability/importability of the resulted files would be helpful.

Reply:
Yes, I think # should be added automatically, which is easy to do.

Reply:
I saw the addition, looks great :)